.github/workflows/javascript-tests.yml

```diff
@@ -0,0 +1,105 @@
+  contents: read
+  deployments: none
+  issues: read
+  packages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+concurrency:
+  group: js-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  Javascript:
+    runs-on: ubuntu-20.04
+    services:
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_DATABASE: piwik_tests
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+        options: --tmpfs /var/lib/mysql:rw
+
+    strategy:
+      fail-fast: false
+      matrix:
+        tests: [ 'JavascriptTests', 'AngularJSTests' ]
+        node: [ '12','16' ]
+        exclude:
+          - tests: 'JavascriptTests'
+            node: '16'
+          - tests: 'AngularJSTests'
+            node: '12'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          lfs: false
+          submodules: recursive
+
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
+          coverage: none
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: installing node/puppeteer
+        working-directory: ./tests/lib/screenshot-testing
+        run: npm install
+
+      - name: installing config.ini.php && Adjusting phpunit.xml && tmp/ sub-directories
+        run: |
+          cp .github/artifacts/config.ini.github.php  config/config.ini.php
+          sed -i 's/3000/\//g' ./config/config.ini.php
+          cp ./tests/PHPUnit/phpunit.xml.dist ./tests/PHPUnit/phpunit.xml
+          mkdir -p ./tmp/assets
+          mkdir -p ./tmp/cache
+          mkdir -p ./tmp/latest
+          mkdir -p ./tmp/logs
+          mkdir -p ./tmp/sessions
+          mkdir -p ./tmp/templates_c
+          mkdir -p ./tmp/templates_c/d2
+          mkdir -p ./tmp/tcpdf
+          mkdir -p ./tmp/climulti
+          chmod a+rw ./tests/lib/geoip-files || true
+          chmod a+rw ./plugins/*/tests/System/processed || true
+          chmod a+rw ./plugins/*/tests/Integration/processed || true
+          sudo chmod -R 755 ./tmp
+
+      - name: Install dependencies
+        run: |
+          composer clear-cache
+          composer install --ignore-platform-reqs
+      - name: Run PHP on 80
+        run: |
+          sudo setcap CAP_NET_BIND_SERVICE=+eip $(readlink -f $(which php))
+          tmux new-session -d -s "php-cgi" sudo php -S 127.0.0.1:80
+          tmux ls
+      - name: Javascript Tests
+        if: matrix.tests == 'JavascriptTests'
+        run: |
+          ./console tests:run-js --matomo-url='http://localhost'
+      - name: Running angularjs && vue tests
+        if: matrix.tests == 'AngularJSTests'
+        working-directory: ./tests/angularjs
+        run: |
+          npm install
+          ./node_modules/karma/bin/karma start karma.conf.js --browsers ChromeHeadless --single-run
```

Comment:
Running the AngularJS tests might not need PHP to be installed at all. We should consider splitting JS & Angular into two actions, so we only set up what is required